components/config/gamesettings.cpp

```diff
@@ -173,6 +174,138 @@ bool Config::GameSettings::writeFile(QTextStream &stream)
+    QString settingLine;
+    QMap<QString, QString>::const_iterator settingsIter = mUserSettings.begin();
+    for (; settingsIter != mUserSettings.end(); ++settingsIter)
+    {
+        settingLine = settingsIter.key()+"="+settingsIter.value();
+        if (settingRegex.indexIn(settingLine) != -1)
+        {
+            userSettingsCopy[settingRegex.cap(1)+"="+settingRegex.cap(2)] =
+                (settingRegex.captureCount() < 3) ? "" : settingRegex.cap(3);
+        }
+    }
+
+    QString keyVal;
+    for (std::vector<QString>::iterator iter = fileCopy.begin(); iter != fileCopy.end(); ++iter)
+    {
+        // skip empty or comment lines
+        if ((*iter).isEmpty() || (*iter).contains(QRegExp("^\\s*#")))
+            continue;
+
+        // look for a key in the line
+        if (settingRegex.indexIn(*iter) == -1 || settingRegex.captureCount() < 2)
+        {
+            // no key or first part of value found in line, replace with a null string which
+            // will be remved later
+            *iter = QString();
+            continue;
+        }
+
+        // look for a matching key in user settings
+        keyVal = settingRegex.cap(1)+"="+settingRegex.cap(2);
+        QMap<QString, QString>::iterator it = userSettingsCopy.find(keyVal);
+        if (it == userSettingsCopy.end())
+        {
+            // no such key+valStart, replace with a null string which will be remved later
+            *iter = QString();
+        }
+        else
+        {
+            *iter = QString(it.key()+it.value());
+            userSettingsCopy.erase(it);
+        }
+    }
+
+    // write the new config file
+    QString key;
+    QString value;
+    for (std::vector<QString>::iterator iter = fileCopy.begin(); iter != fileCopy.end(); ++iter)
+    {
+        if ((*iter).isNull())
+            continue;
+
+        // Below is based on readFile() code, if that changes corresponding change may be
+        // required (for example duplicates may be inserted if the rules don't match)
+        if ((*iter).isEmpty() || (*iter).contains(QRegExp("^\\s*#")))
+        {
+            stream << *iter << "\n";
+            continue;
+        }
+
+        if (settingRegex.indexIn(*iter) == -1 || settingRegex.captureCount() < 2)
+            continue;
+
+        // Quote paths with spaces
+        key = settingRegex.cap(1);
+        value = settingRegex.cap(2)+settingRegex.cap(3);
+        if (key == QLatin1String("data")
+            || key == QLatin1String("data-local")
+            || key == QLatin1String("resources"))
+        {
+            if (value.contains(QChar(' ')))
+            {
+                value.remove(QChar('\"')); // Remove quotes
+
+                stream << key << "=\"" << value << "\"\n";
+                continue;
+            }
+        }
+        stream << key << "=" << value << "\n";
+    }
+
+    if (!userSettingsCopy.empty())
+    {
+        stream << "# new entries" << "\n";
```

Comment:
What is the point of this comment?

I noticed that my openmw.cfg now contains
<pre>
# new entries
# new entries

</pre>

at the end, and I fear that more of these will accumulate every time that I use the launcher, or on some other event. Can we remove it please?